ofrak_core/src/ofrak/core/lz4.py

```diff
@@ -0,0 +1,332 @@
+    """
+
+    targets = (Lz4ModernData, Lz4SkippableData)
+    children = (GenericBinary,)
+
+    async def unpack(self, resource: Resource, config=None):
+        """
+        Unpack LZ4 data.
+
+        :param resource: The LZ4 resource to unpack
+
+        :raises RuntimeError: if the data is not valid LZ4 format
+        """
+        resource_data = await resource.get_data()
+
+        if resource.has_tag(Lz4ModernData):
+            # lz4.frame.get_frame_info() does not support legacy frames
+            frame_info = lz4.frame.get_frame_info(resource_data)
+            resource.add_view(
+                Lz4ModernData(
+                    block_size=frame_info["block_size"],
+                    block_size_id=frame_info["block_size_id"],
+                    block_linked=frame_info["block_linked"],
+                    content_checksum=frame_info["content_checksum"],
+                    block_checksum=frame_info["block_checksum"],
+                    content_size=frame_info["content_size"],
+                )
+            )
+
+        try:
+            decompressed_data = lz4.frame.decompress(resource_data)
+        except RuntimeError as e:
+            LOGGER.error(f"Failed to decompress LZ4 data: {e}")
+            raise
+
+        await resource.create_child(
+            tags=(GenericBinary,),
+            data=decompressed_data,
+        )
+
+
+class Lz4LegacyUnpacker(Unpacker[None]):
+    """
+    Unpack (decompress) LZ4 legacy frame format files.
+
+    Legacy format (v0.1-v0.9) uses lz4.block decompression instead of lz4.frame.
+    """
+
+    targets = (Lz4LegacyData,)
+    children = (GenericBinary,)
+
+    async def unpack(self, resource: Resource, config=None):
+        """
+        Unpack LZ4 legacy data.
+
+        :param resource: The LZ4 legacy resource to unpack
+
+        :raises RuntimeError: if the data is not valid LZ4 legacy format
+        """
+        resource_data = await resource.get_data()
+        if len(resource_data) < 8:
+            raise RuntimeError("Invalid LZ4 legacy format: file too short")
+        deserializer = BinaryDeserializer(
+            io.BytesIO(resource_data),
+            endianness=Endianness.LITTLE_ENDIAN,
+            word_size=4,
+        )
+        magic = deserializer.read(4)
+        assert magic == LZ4_LEGACY_MAGIC
+
+        decompressed_data = b""
+        while deserializer.position() < len(resource_data):
+            # Legacy LZ4 has a repeating pattern of 4 bytes block size followed by compressed data
+            block_size = deserializer.unpack_uint()
+            compressed_block = deserializer.read(block_size)
+            try:
+                # LZ4 legacy block size (uncompressed) is 8 MB (see https://github.com/lz4/lz4/blob/67a385a170d2dc331a25677e0d20d96eef0450c5/programs/lz4io.c#L86)
+                decompressed_data += lz4.block.decompress(
+                    compressed_block,
+                    uncompressed_size=8 * (1 << 20),
```

Comment:
```suggestion
                    uncompressed_size=8 * 1024 * 1024,
```

Sometimes I find it clearer to express MB in terms of KB squared. Feel free to ignore this – just a matter of taste.

Reply:
For context I chose to use the same syntax that the `lz4` repo used here. I agree it's less intuitive, but wanted to ensure it matches the reference implementation.